x-pack/plugins/security_solution/public/common/components/threat_match/helpers.tsx

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  ThreatMap,
+  threatMap,
+  ThreatMapping,
+} from '../../../../common/detection_engine/schemas/types';
+
+import { IIndexPattern, IFieldType } from '../../../../../../../src/plugins/data/common';
+import { Entry, FormattedEntry, ThreatMapEntries, EmptyEntry } from './types';
+
+/**
+ * Formats the entry into one that is easily usable for the UI.
+ *
+ * @param patterns IIndexPattern containing available fields on rule index
+ * @param item item entry
+ * @param itemIndex entry index
+ */
+export const getFormattedEntry = (
+  indexPattern: IIndexPattern,
```

Comment:
I know this is historical to the OG `and`/`or` implementation, but just as an FYI there has been movement away from using `IIndexPattern` as it's confusing `IndexPattern` and data objects. https://github.com/elastic/kibana/pull/77968

Reply:
I see I should be using `IndexPattern` instead. I will try it out.

I wonder why there's not a deprecated above it yet or if maybe it's not deprecated. fwiw, the other deprecations show up in my IDE as _very_ useful to prefer usage to something else.

Reply:
Ok, was able to bubble this everywhere. I had to do some "as casting" due to legacy parts of the app but let's see how it goes with this.